src/mdns/mdns.cpp

```diff
@@ -39,54 +39,39 @@ namespace otbr {
 
 namespace Mdns {
 
-bool Publisher::IsServiceTypeEqual(const char *aFirstType, const char *aSecondType)
+bool Publisher::IsServiceTypeEqual(std::string aFirstType, std::string aSecondType)
 {
-    size_t firstLength  = strlen(aFirstType);
-    size_t secondLength = strlen(aSecondType);
-
-    if (firstLength > 0 && aFirstType[firstLength - 1] == '.')
+    if (!aFirstType.empty() && aFirstType.back() == '.')
     {
-        --firstLength;
+        aFirstType.pop_back();
     }
-    if (secondLength > 0 && aSecondType[secondLength - 1] == '.')
+
+    if (!aSecondType.empty() && aSecondType.back() == '.')
     {
-        --secondLength;
+        aSecondType.pop_back();
     }
 
-    return firstLength == secondLength && memcmp(aFirstType, aSecondType, firstLength) == 0;
+    return aFirstType == aSecondType;
 }
 
-otbrError Publisher::EncodeTxtData(const TxtList &aTxtList, uint8_t *aTxtData, uint16_t &aTxtLength)
+otbrError Publisher::EncodeTxtData(const TxtList &aTxtList, std::vector<uint8_t> &aTxtData)
```

Comment:
nit: Use `vector::reserve` to reduce the times of memory allocations.

Reply:
May not easy to estimate the size, I think `vector::reserve` has its advantages only if we already know the target size this vector is going to grow to.